app/views/admin/attachments/_form.html.erb

```diff
@@ -32,12 +48,28 @@
       <% end %>
     <% end %>
   <% elsif attachment.is_a?(ExternalAttachment) %>
-    <%= form.text_field :external_url %>
+    <div class="govuk-!-margin-bottom-8">
+      <%= render "govuk_publishing_components/components/input", {
+        label: {
+          text: "External url (required)"
+        },
+        name: "attachment[external_url]",
+        heading_level: 2,
+        heading_size: "l",
+        value: form.object.external_url
+      } %>
+    </div>
   <% else %>
     <%= render 'attachment_data_fields', form: form %>
   <% end %>
 
   <%= hidden_field_tag :type, params[:type] %>
 
-  <%= form.save_or_cancel cancel: attachable_attachments_path(attachable) %>
+  <% if attachment.is_a?(ExternalAttachment) %>
+    <%= render "govuk_publishing_components/components/button", {
+      text: "Next"
+    } %>
+  <% else %>
```

Comment:
Did some new logic get added here? Looking at the legacy file, it seems like the save_or_cancel always shows.

Reply:
The idea is to keep `save_or_cancel` when the attachment is anything other than an external attachment, in which case it uses the design system `next` button. Either way it submits the form.